src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -8114,22 +8206,30 @@ void StorageReplicatedMergeTree::movePartitionToTable(const StoragePtr & dest_ta
         Coordination::Stat alter_partition_version_stat;
         zookeeper->get(alter_partition_version_path, &alter_partition_version_stat);
 
-        MergeTreePartInfo drop_range;
-        std::optional<EphemeralLockInZooKeeper> delimiting_block_lock;
-        getFakePartCoveringAllPartsInPartition(partition_id, drop_range, delimiting_block_lock, true);
-        String drop_range_fake_part_name = getPartNamePossiblyFake(format_version, drop_range);
+        MergeTreePartInfo source_drop_range;
+        std::optional<EphemeralLockInZooKeeper> source_delimiting_block_lock;
+        getFakePartCoveringAllPartsInPartition(source_partition_id, source_drop_range, source_delimiting_block_lock, true);
+        String source_drop_range_fake_part_name = getPartNamePossiblyFake(format_version, source_drop_range);
 
         DataPartPtr covering_part;
         DataPartsVector src_all_parts;
         {
             /// NOTE: Some covered parts may be missing in src_all_parts if corresponding log entries are not executed yet.
             auto parts_lock = src_data.lockParts();
-            src_all_parts = src_data.getActivePartsToReplace(drop_range, drop_range_fake_part_name, covering_part, parts_lock);
+            src_all_parts = src_data.getActivePartsToReplace(source_drop_range, source_drop_range_fake_part_name, covering_part, parts_lock);
         }
 
         if (covering_part)
             throw Exception(ErrorCodes::LOGICAL_ERROR, "Got part {} covering drop range {}, it's a bug",
-                            covering_part->name, drop_range_fake_part_name);
+                            covering_part->name, source_drop_range_fake_part_name);
+
+        // TODO fix below
```

Comment:
Ideally, this would be done only once (as it is being done in other methods `replacePartitionFrom` and `movePartitionToTable` on non-replicated engine. The problem is that in order to verify compatibility between partitions and calculate new partition id, the source parts are required. Source parts are grabbed only inside the loop and I am still not sure if this logic can be moved out of here.